### ISSUE TITLE
tasks not awaited when closing the connection

### ISSUE THREAD

**Issue by zsorizs** [frankie567/httpx-ws#49]
## Describe the bug

In our tests we were getting warnings that tasks _background_receive_task and _background_keepalive_ping_task were terminated while pending.

## Expected behavior

All tasks should be awaited to avoid warnings.

## Configuration
 - Python version: 3.11
 - httpx-ws version: 0.4.1

## Possible fix
The next patch fixes the issue for us:
```
diff --git a/httpx_ws/_api.py b/httpx_ws/_api.py
index 5abfebe..cb51a34 100644
--- a/httpx_ws/_api.py
+++ b/httpx_ws/_api.py
@@ -924,6 +924,7 @@ class AsyncWebSocketSession:
             except httpcore.WriteError:
                 pass
         await self.stream.aclose()
+        await asyncio.gather(self._background_receive_task, self._background_keepalive_ping_task)

     async def _background_receive(self, max_bytes: int) -> None:
         """
```

**Comment by frankie567** [comment]
So, do you confirm it happens when using the `ASGIWebSocketTransport`?

**Comment by zsorizs** [comment]
> So, can you confirm it happens when using the `ASGIWebSocketTransport`?

Yes, I'm using that transport.

**Comment by frankie567** [comment]
I think this is fixed indirectly by #58
